docs/data-sources/signed_pipeline_steps.md

```diff
@@ -0,0 +1,94 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "buildkite_signed_pipeline_steps Data Source - terraform-provider-buildkite"
+subcategory: ""
+description: |-
+  Use this data source to sign pipeline steps with a JWKS key. You will need to have the
+  corresponding verification key present on the agents that run this the steps in this
+  pipeline. You can use then use these steps in a buildkite_pipeline resource.
+  ~> Security Notice The secret key required to use this data source will be stored
+  unencrypted in your Terraform state file. If you wish to avoid this, you can manually
+  sign the pipeline line steps using the buildkite agent CLI https://buildkite.com/docs/agent/v3/cli-tool#sign-a-step.
+  Example Usage
+  ```terraform
+  locals {
+    repository = "git@github.com:my-org/my-repo.git"
+  }
+  resource "buildkitepipeline" "my-pipeline" {
+    name       = "my-pipeline"
+    repository = local.repository
+    steps      = data.buildkitesignedpipelinesteps.my-signed-steps.steps
+  }
+  data "buildkitesignedpipelinesteps" "my-signed-steps" {
+    repository  = local.repository
+    jwks        = file(var.jwksfile)
+    jwkskeyid = var.jwkskeyid
+  unsigned_steps = <<YAML
+  steps:
+  - label: ":pipeline:"
+    command: buildkite-agent pipeline upload
+  YAML
+  }
+  ```
+  More info in the Buildkite documentation https://buildkite.com/docs/agent/v3/signed_pipelines.
+---
+
+# buildkite_signed_pipeline_steps (Data Source)
+
+Use this data source to sign pipeline steps with a JWKS key. You will need to have the
+corresponding verification key present on the agents that run this the steps in this
+pipeline. You can use then use these steps in a buildkite_pipeline resource.
+
+~> **Security Notice** The secret key required to use this data source will be stored
+*unencrypted* in your Terraform state file. If you wish to avoid this, you can manually
+sign the pipeline line steps using the [buildkite agent CLI](https://buildkite.com/docs/agent/v3/cli-tool#sign-a-step).
+
+## Example Usage
+```terraform
+locals {
+  repository = "git@github.com:my-org/my-repo.git"
+}
+
+resource "buildkite_pipeline" "my-pipeline" {
+  name       = "my-pipeline"
+  repository = local.repository
+  steps      = data.buildkite_signed_pipeline_steps.my-signed-steps.steps
```

Comment:
i had just been typing up the comment "i think we should add an example here", and you beat me to the punch!

i wonder if given the stuttering here, we might want to call the output of the signed_pipeline_steps data source `output` or `yaml` or something... `yaml` being cognate to the `aws_iam_policy_document` data source's `yaml` output

Reply:
Hmm, I had thought b/c the `pipeline` resource's input was `steps`, we should make the output of this `steps` as well, so it's clear without documentation how to connect the two.